platforms/core-configuration/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/GradleKotlinDslIntegrationTest.kt

```diff
@@ -214,9 +214,8 @@ class GradleKotlinDslIntegrationTest : AbstractKotlinIntegrationTest() {
         reason = "Class path isolation, tested here, is not correct in embedded mode"
     )
     fun `can compile against a different (but compatible) version of the Kotlin compiler`() {
-
-        val differentKotlinVersion = "1.6.0"
-        val expectedKotlinCompilerVersionString = "1.6.0"
+        val differentKotlinVersion = KotlinGradlePluginVersions().latestsStable.last { it != embeddedKotlinVersion }
```

Comment:
❓ This probably needs a check from @gradle/bt-kotlin-dsl-maintainers

Reply:
`latestsStable` might contain a future incompatible version.
It would be more correct and future-proof to get the latest stable version that is lower than `embeddedKotlinVersion ` instead.